### ISSUE TITLE
'yield-star-spacing' rule conflict 

### ISSUE THREAD

**Issue by xianghongai** [eslint/eslint#11448]
**Tell us about your environment**

* **ESLint Version:5.14.1**
* **Node Version:8.15.0**
* **npm Version:6.8.0**

**What parser (default, Babel-ESLint, etc.) are you using?**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
  "root": true,
  "extends": [
    "standard",
    "prettier",
    "prettier/standard"
  ],
  "plugins": [
    "standard",
    "prettier"
  ],
  "parserOptions": {
    "ecmaVersion": 9,
    "sourceType": "module",
    "ecmaFeatures": {}
  },
  "globals": {
    "_": "readonly"
  },
  "rules": {
    "generator-star-spacing": [ "warn", "before" ],
    "yield-star-spacing": [ "warn", "before" ]
  }
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
// yield-star-spacing.js
/* eslint yield-star-spacing: ["error", "after"] */
function* generator() { yield* other(); }

/* eslint yield-star-spacing: ["error", "before"] */
function *generator2() { yield *other(); }

/* eslint yield-star-spacing: ["error", "both"] */
function *generator3() { yield * other(); }

/* eslint yield-star-spacing: ["error", "neither"] */
function*generator4() { yield*other(); }

function other () {}
```

```bash
./node_modules/.bin/eslint ./src/yield-star-spacing.js
```

**What did you expect to happen?**
expect of the  `eslint yield-star-spacing: ["error", "before"]`  rule.

**What actually happened? Please include the actual, raw output from ESLint.**
![eslint-yield-star-spacing](https://user-images.githubusercontent.com/2075006/53475119-1d2f4800-3aaa-11e9-9d95-4bfd8ad6b2a9.png)

**Are you willing to submit a pull request to fix this bug?**
Sorry, I can't do it.

**Comment by g-plane** [comment]
Hi @xianghongai , thanks for this issue.

What conflicts with the `generator-star-spacing` have you encountered?

**Comment by xianghongai** [comment]
Hi, @g-plane ,I have updated the 'Configuration' section of this issue,

https://github.com/Microsoft/vscode-eslint/issues/630

**Comment by g-plane** [comment]
Can you state what the conflict is?

**Comment by xianghongai** [comment]
1. config
```json
"rules": {
    "generator-star-spacing": [ "warn", "before" ],
    "yield-star-spacing": [ "warn", "before" ]
  }
```

2. js
```javascript
/* eslint yield-star-spacing: ["error", "after"] */
function* generator() { yield* other(); }

/* eslint yield-star-spacing: ["error", "before"] */
function *generator2() { yield *other(); }

/* eslint yield-star-spacing: ["error", "both"] */
function * generator3() { yield * other(); }

/* eslint yield-star-spacing: ["error", "neither"] */
function*generator4() { yield*other(); }

function other () {}
```

3. result

```javascript
function *generator2() { yield *other(); }
```

the code `yield *other();`(line: 5) prompt "`Unexpected space before *    yield-star-spacing`"

for more information refer to the images in the subject, and compare the left arrow and right arrow,

thx~

**Comment by xianghongai** [comment]
![yss](https://user-images.githubusercontent.com/2075006/53536005-9d53bd00-3b3f-11e9-8fda-d9917ca7fa92.png)
